### DISCUSSION TITLE
Graphing CDM vocabulary

### DISCUSSION THREAD

**Discussion by kangyh9659** [callahantiff/PheKnowLator#132] [Q&A]
Hello, Tiffany and all.
First of all, after I went through a tutorial ipynb file, I thought this stuff is awesome! I think this would be very helpful to visualize graph for various concepts.
Well,, I'm sorry to bother you but I was wondering how to utilize the graphics for the OMOP CDM vocabulary..
I want to use pheknowlator/notebook/tutorials/entity_search/entity_search.ipynb file to do that... or any other existing ipynb file is fine.
I just was wondering where to put the OMOP CDM concept id possibly since the coding is extensively complicated.
If you could help me with start, I would very appreciate so much.
By the way, I fixed the previous problem by adding `from pkt_kg.utils.kg_utils import *` on the coding in entity_search.ipynb file. Thank you for help previously.

UPDATE:
Or do I need to utilize [OMOP2OBO] for OMOP CDM VOCA? 

Thanks again for your help.

**Comment by callahantiff** [comment]
Hey @kangyh9659!

Happy to help! This script is actually still in process, so I apologize for it not being in better shape. I am going to be working on it a lot over the next two weeks though. There is a piece that I have not yet integrated, which provides the functionality that you are asking about. It's called OMOP2OBO.  If you are not in a hurry and are OK with waiting a few days, can I reach back out to you the end of next week with some updates. Would that be Ok?

Thanks again for your feedback!

-Tiffany

**Reply by kangyh9659** [reply]
Hi Tiffany,
If you could integrate OMOP2OBO, I would be more than happy to celebrate with you because I could start using this awesome stuff! I'm looking forward to use it so much. The reason why I need this stuff so much is that in the oracle setting, one can try to code for ancestor-descendant hierarchy, but it sometimes happens that endless loop is formed because of way the concept relationships are defined and the code fails. Hope to see the graph of concepts hierarchy in your stuff. Thank you again for your help! Hope you have a wonderful weekend!

**Reply by callahantiff** [reply]
Perfect! Your project sounds interesting, thanks for sharing some information about it! 

Just to verify, is your goal to to search the hierarchy of the OMOP standard terminologies more efficiently? So, you would use OMOP2OBO to map into the OBO Foundry ontologies, find relevant concepts by searching their hierarchies and then map the resulting codes back to OMOP?

**Reply by kangyh9659** [reply]
First of all thanks for update. I love that you helps me in a specific way. If I share more, I'm having an ongoing project of analyzing characterisitics of non small cell lung cancer paitents, and in there, for example, one can want to graph hierarchy of several concepts of condition/procedure/drugs whatnot. And specifically, for condition, not only SNOMED code but also ICD code hierachy might be preferred, too. But this is just an example. Most importantly, by looking at the hierachy, one can grasp the 'descendant' concepts that lies in each level of hierachy. Since specific descendant might not lie in the most bottom level, one might want to see the graph of the hierachy. For example, from the ancestor concept 'malignant neoplastic disease', one might want to look at specific cancers. The 'Malignant neoplasm of respiratory tract' or 'malignant neoplasm of abdomen' descendants might be too broad. Instead, one might want to look for 'malgnant neoplasm of liver', or 'Malignant tumor of lung' which lies within different levels of hierarchy because of way the SNOMED hierarchy is defined. I'm not a physician so, I'm not familiar of tumors, so just looking at the graph might be helpful. Again, thanks for your help and hope you have a nice day.

**Reply by callahantiff** [reply]
Hi @kangyh9659 -- so sorry for the long delay in getting back to you. Thanks so much for sharing detailed information on your project. It sounds really interesting! I'm so thrilled that you are exploring PheKnowLator and hope that there are things in the tutorial that might be able to help you solve some of the challenges that you describe.

I have (finally 🙀) finished making the changes we discussed to the original tutorial as well as some other fun enhancements which includes improved (and interactive, yay!) graphics as well as a new script that is run from the notebook to automatically download the `OMOP2OBO` mappings (initially just for condition occurrence concepts and drug exposure ingredients) and convert them into a dictionary, which is then used as part of the tutorial workflow. I also added simple paths to the path-based search section as I thought this might also be useful to your use case.

I really appreciate your discussion and hope that these changes will be useful to you. They will definitely require a full update of the Master branch (many files were modified). Would love to know what you think of the changes. 

Best of luck with everything, I'd love to stay in touch about your project and what solutions you ultimately come up with! 😄

**Reply by kangyh9659** [reply]
Hello, @callahantiff . Thank you so much for your so much effort on this. I really appreciate your help. I'm sure it will be used in our research valuably many times. I'll try to update our research and how we used your pacakge. Thanks again! Best of luck with everything on your works.

**Reply by callahantiff** [reply]
@kangyh9659 - That sounds fantastic, I would be very interested to hear what you think after you have had time to explore it. Thank you again for your feedback. best of luck with your work! 😄
